### PR TITLE
Feature/conjunction generation

### PR DESCRIPTION
**Pull request by kilativ-dotcom**

### CODE REVIEW

File: problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp [hunk 1]
```diff
@@ -79,6 +79,7 @@ LogicFormulaResult TemplateExpressionNode::generate(Replacements & replacements)
   LogicFormulaResult result;
   result.isGenerated = false;
   std::vector<ScTemplateParams> paramsVector = ReplacementsUtils::getReplacementsToScTemplateParams(replacements);
+  std::set<std::string> varNames = templateSearcher->getVarNames(formulaTemplate);
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp:82]:
```suggestion
  std::set<std::string> const & varNames = templateSearcher->getVarNames(formulaTemplate);
```

File: problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp [hunk 2]
```diff
@@ -103,7 +104,23 @@ LogicFormulaResult TemplateExpressionNode::generate(Replacements & replacements)
       {
         result.isGenerated = true;
         result.value = true;
-        result.replacements = replacements;
+        Replacements temporalReplacements;
+        for (const auto & name: varNames)
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp:108]:
```suggestion
        for (std::string const & name: varNames)
```

File: problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp [hunk 3]
```diff
@@ -103,7 +104,23 @@ LogicFormulaResult TemplateExpressionNode::generate(Replacements & replacements)
...
+        {
+          ScAddrVector replacementsVector;
+          bool generationHasVar =
+              generationResult.GetReplacements().find(name) != generationResult.GetReplacements().end();
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp:112]:
```suggestion
              generationResult.GetReplacements().find(name) != generationResult.GetReplacements().cend();
```

File: problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp [hunk 4]
```diff
@@ -103,7 +104,23 @@ LogicFormulaResult TemplateExpressionNode::generate(Replacements & replacements)
       {
         result.isGenerated = true;
         result.value = true;
-        result.replacements = replacements;
+        Replacements temporalReplacements;
+        for (const auto & name: varNames)
+        {
+          ScAddrVector replacementsVector;
+          bool generationHasVar =
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp:111]:
```suggestion
          bool const generationHasVar =
```

File: problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp [hunk 5]
```diff
@@ -103,7 +104,23 @@ LogicFormulaResult TemplateExpressionNode::generate(Replacements & replacements)
...
+              generationResult.GetReplacements().find(name) != generationResult.GetReplacements().end();
+          ScAddr outResult;
+          bool paramsHaveVar = scTemplateParams.Get(name, outResult);
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp:114]:
```suggestion
          bool const paramsHaveVar = scTemplateParams.Get(name, outResult);
```

File: problem-solver/cxx/inferenceModule/test/units/TestComplexFormulas.cpp [hunk 6]
```diff
@@ -304,4 +304,34 @@ TEST_F(InferenceComplexFormulasTest, EquivalencesNested)
...
 }
 
+// (a -> (b /\ c))
+TEST_F(InferenceComplexFormulasTest, ConjunctionInConclusionTest)
+{
+  ScMemoryContext context(sc_access_lvl_make_min, "testComplexConclusion");
+
+  string const & name = TEST_FILES_DIR_PATH + "conjunctionInConclusionTest.scs";
+  struct stat buffer;
+  SC_LOG_DEBUG("exists = " + to_string(stat(name.c_str(), &buffer) == 0));
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/test/units/TestComplexFormulas.cpp:314]: Remove it

File: problem-solver/cxx/inferenceModule/test/testStructures/LogicModule/ComplexFormulas/conjunctionInConclusionTest.scs [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    -> argument;;
+
+inference_logic_test_question
+    -> rrel_1: target;
+    -> rrel_2: rules_set;
+    -> rrel_3: arguments;;
+
+
+
+
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/test/testStructures/LogicModule/ComplexFormulas/conjunctionInConclusionTest.scs:64]: Remove new lines

File: problem-solver/cxx/inferenceModule/logic/ConjunctionExpressionNode.cpp [hunk 8]
```diff
@@ -99,3 +99,20 @@ LogicFormulaResult ConjunctionExpressionNode::compute(LogicFormulaResult & resul
   }
   return result;
 }
+
+LogicFormulaResult ConjunctionExpressionNode::generate(Replacements & replacements) const
+{
+  LogicFormulaResult fail = {false, false, {}};
+  LogicFormulaResult globalResult = {true, false, replacements};
+  for (const auto & operand: operands)
```
Comment (NikitaZotov) [inline problem-solver/cxx/inferenceModule/logic/ConjunctionExpressionNode.cpp:107]:
```suggestion
  for (auto const & operand: operands)
```

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: problem-solver/cxx/inferenceModule/logic/ConjunctionExpressionNode.cpp (hunk 8 above)
Comment (MksmOrlov) [inline problem-solver/cxx/inferenceModule/logic/ConjunctionExpressionNode.cpp:107]:
```suggestion
  for (auto const & operand : operands)
```

File: problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp (hunk 2 above)
Comment (MksmOrlov) [inline problem-solver/cxx/inferenceModule/logic/TemplateExpressionNode.cpp:108]:
```suggestion
        for (auto const & name : varNames)
```

**Review by MksmOrlov — APPROVED** [review]

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**